### PR TITLE
add PrecompiledError

### PR DESCRIPTION
**Pull request by mingzhenliu**
1、add PrecompiledError
2、change namespace
3、fix bug

### CODE REVIEW

File: libevm/ExtVMFace.h [hunk 1]
```diff
@@ -32,14 +32,19 @@
 #include <libethcore/LastBlockHashesFace.h>
 #include <libethcore/LogEntry.h>
 
-#include <evmc/evmc.h>
+#include <evmc/include/evmc/evmc.h>
```
Comment (wheatli) [inline libevm/ExtVMFace.h]: just use <evmc/evmc.h>

File: libexecutivecontext/Executive.cpp [hunk 2]
```diff
@@ -172,49 +172,23 @@ bool Executive::call(CallParameters const& _p, u256 const& _gasPrice, Address co
...
-                if (!success)
-                {
-                    m_gas = 0;
-                    m_excepted = TransactionException::OutOfGas;
-                    return true;  // true means no need to run go().
-                }
-            }
-        }
-        else//*/
+    /* if (m_envInfo.precompiledEngine()->isOrginPrecompiled(_p.codeAddress))
```
Comment (wheatli) [inline libexecutivecontext/Executive.cpp]: why commented?

File: libethcore/TransactionReceipt.cpp [hunk 3]
```diff
@@ -0,0 +1,74 @@
+/*
```
Comment (wheatli) [inline libethcore/TransactionReceipt.cpp]: conflict with history commits

File: libblockverifier/ExecutiveContextFactory.cpp [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+#include <libstorage/MemoryTableFactory.h>
+#include <libstorage/TableFactoryPrecompiled.h>
+
+using namespace dev;
+using namespace dev::blockverifier;
+
+void ExecutiveContextFactory::initExecutiveContext(
+    BlockInfo blockInfo, ExecutiveContext::Ptr context)
+{
+#if 0
```
Comment (wheatli) [inline libblockverifier/ExecutiveContextFactory.cpp]: drop the unused codes

File: libblockverifier/ExecutiveContext.cpp [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+}
+
+bool ExecutiveContext::isOrginPrecompiled(Address const& _a) const
+{
+    return m_precompiledContract.count(_a);
+}
+
+bigint ExecutiveContext::costOfOrginPrecompiled(Address const& _a, bytesConstRef _in) const
+{
+    return m_precompiledContract.at(_a).cost(_in);
```
Comment (wheatli) [inline libblockverifier/ExecutiveContext.cpp]: if the address is not exist...

File: libblockverifier/BlockVerifier.cpp [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+#include <libethcore/PrecompiledContract.h>
+#include <libethcore/TransactionReceipt.h>
+using namespace dev;
+using namespace dev::eth;
+using namespace dev::blockverifier;
+
+ExecutiveContext::Ptr BlockVerifier::executeBlock(Block& block, int stateType,
+    std::unordered_map<Address, PrecompiledContract> const& precompiledContract)
+{
+    /* LOG(TRACE) << "Block:enact tx_num=" << block.transactions().size();
```
Comment (wheatli) [inline libblockverifier/BlockVerifier.cpp]: we need give out the implementation and ut

File: libexecutivecontext/Executive.cpp (hunk 2 above)
Comment (mingzhenliu) [reply libexecutivecontext/Executive.cpp]: Will be implemented later

File: libblockverifier/ExecutiveContext.cpp (hunk 5 above)
Comment (mingzhenliu) [reply libblockverifier/ExecutiveContext.cpp]: isOrginPrecompiled will check

File: libblockverifier/BlockVerifier.cpp (hunk 6 above)
Comment (mingzhenliu) [reply libblockverifier/BlockVerifier.cpp]: Will be added later

**Pull request merged**